extensions/amp-form/0.1/amp-form.js

```diff
@@ -155,14 +168,40 @@ export class AmpForm {
     /** @const @private {!./form-validators.FormValidator} */
     this.validator_ = getFormValidator(this.form_);
 
-    // TODO(mkhatib, #6927): Wait for amp-selector to finish loading if the current form
-    // is using it.
-    this.actions_.installActionHandler(
-        this.form_, this.actionHandler_.bind(this));
+    // Wait for amp-selector to finish loading if the current form is using it.
+    this.whenDependenciesReady_().then(() => {
+      this.actions_.installActionHandler(
+          this.form_, this.actionHandler_.bind(this));
+    });
     this.installEventHandlers_();
   }
 
   /**
+   * Returns a promise that will be resolved when all dependencies used inside the form
+   * tag are loaded (e.g. amp-selector) or 500ms timeout - whichever is first.
+   * @return {!Promise}
+   * @private
+   */
+  whenDependenciesReady_() {
+    const depsDict = {};
+    const depsPromises = [];
+    for (let i = EXTERNAL_DEPS.length - 1; i >= 0; i--) {
+      const tagName = EXTERNAL_DEPS[i];
+      const depIsUsed = this.form_.getElementsByTagName(tagName).length !== 0;
+      if (depIsUsed && depsDict[tagName] === undefined) {
+        // TODO(mkhatib, #7032): Implement a way to wait for an element to be built to
```

Comment:
I think that the implementation would be drastically different and we should consider doing build promise work first.